src/components/spreadsheet/use-spreadsheet-equipments.ts

```diff
@@ -144,7 +159,80 @@ export const useSpreadsheetEquipments = (
+            let additionalEquipmentsByNodes: Record<string, Record<SpreadsheetEquipmentType, Identifiable[]>> = {};
+            if (spreadsheetEquipmentType) {
+                customColumns.forEach((column) => {
+                    const pattern: string = aliases.map((alias) => `\\b${alias}\\b\\.`).join('|'); // pattern to find the aliases in the expression
+                    const regex = new RegExp(pattern, 'g');
+                    const aliasesFoundInFormula = column.formula?.match(regex);
+                    aliasesFoundInFormula?.forEach((aliasWithDot) => {
+                        const alias = aliasWithDot.slice(0, -1); //because we searched for "alias." with the regex
+                        //To avoid fetching multiple time the same data because it's used in multiple column
+                        if (!aliasesAlreadyFound.includes(alias)) {
+                            aliasesAlreadyFound.push(alias);
+                            const nodeNameAssociatedToAlias = customColumnsNodesAliases.find(
+                                (nodeAlias) => nodeAlias.alias === alias
+                            )?.name;
+                            const nodeIdAssociatedToAlias = treeModel?.treeNodes?.find(
+                                (treeNode) => treeNode.data.label === nodeNameAssociatedToAlias
+                            )?.id;
+                            if (nodeIdAssociatedToAlias) {
+                                const fetcherPromises = getFetchers(spreadsheetEquipmentType).map((fetcher) =>
+                                    fetcher(studyUuid, nodeIdAssociatedToAlias, [])
+                                );
+                                fetchers.push(fetcherPromises[0]);
+                                fetcherPromises[0].then((res) => {
+                                    let fetchedEquipments = res.flat();
+                                    if (formatFetchedEquipments) {
+                                        fetchedEquipments = formatFetchedEquipments(fetchedEquipments);
+                                        let fetchedEquipmentByType: Record<SpreadsheetEquipmentType, Identifiable[]> =
+                                            {} as Record<SpreadsheetEquipmentType, Identifiable[]>;
+                                        fetchedEquipmentByType[spreadsheetEquipmentType] = fetchedEquipments;
+                                        additionalEquipmentsByNodes = { ...additionalEquipmentsByNodes };
+                                        additionalEquipmentsByNodes[alias] = fetchedEquipmentByType;
+                                    }
+                                });
+                            }
+                        }
+                    });
+                });
+
+                //so we only dispatch once all the fetches are over
+                Promise.all(fetchers).then(() => {
+                    dispatch(addAdditionalEquipmentsByNodesForCustomColumns(additionalEquipmentsByNodes));
```

Comment:
Why do you create additional data in the redux for this ? Why not directly in the spreadsheetNetwork ?

Reply:
I thought it's better to have the other nodes info in a seperate state because its probable that it won't behave exactly as the current data info

Reply:
I think that's the whole point, it should behave exactly the same. So that, we just load more data initially and then nothing else changes in the code.

Reply:
ok we'll wee that in another ticket